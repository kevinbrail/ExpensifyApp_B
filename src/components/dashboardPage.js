import React from 'react';
import ExpenseList from './expenseList'
import ExpenseListFilters from './expenseListFilters'
import ExpenseSummary from './expenseSummary'

const ExpenseDashboardPage =() => (
    <div>
    <ExpenseSummary />
    <ExpenseListFilters />
    <ExpenseList />
    </div>
)

export default ExpenseDashboardPage;