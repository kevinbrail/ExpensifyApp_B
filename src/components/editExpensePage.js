import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from "./expenseForm"
import { startEditExpense } from '../actions/expenses'
import { startRemoveExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
        //this.props.dispatch(editExpense(this.props.expense.id, expense));
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/')
       // console.log(expense);
    };
    onRemove = (expense) => {
         //this.props.dispatch(removeExpense({id: this.props.expense.id}));
         this.props.startRemoveExpense({id: this.props.expense.id});
         this.props.history.push('/')
    };
    render() {
        return (
            <div>
            <div className="page-header">
                <div className="content-container">
                    <h1 className="page-header__title">Edit Expense</h1>
                </div>    
            </div>
            <div className="content-container">
                <ExpenseForm 
                    expense={this.props.expense}
                    onSubmit={this.onSubmit}
                />
                <button 
                    onClick={this.onRemove}
                    className="button button--secondary">Remove Expense</button>
                </div>
            </div>      
        )}

}

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    }
};
  const mapDispatchToProps = (dispatch, props) => ({
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data)) 
  });

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);