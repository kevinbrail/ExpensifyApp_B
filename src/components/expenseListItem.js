import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem = ({description, amount, note, createdAt, id }) => (
         <Link className="list-item" to={`/edit/${id}`}>
         <div>
            <h3 className="list-item__title">{description}</h3>
            <span className="list-iten__subtitle">
            {note}<br/>
            {moment(createdAt).format('MMMM Do, YYYY')} </span>
         </div>
         <div>
            <h3 className="list-item__data">{numeral(amount / 100).format('$0,00.00')}</h3>
         </div>
        </Link>
);

export default ExpenseListItem;