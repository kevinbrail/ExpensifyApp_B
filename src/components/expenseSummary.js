import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import totalExpenses from '../selectors/expenses-total'
import selectExpenses from '../selectors/expenses';
import {Link} from 'react-router-dom';

export const ExpenseSummary = ({expensesCount, expensesTotal}) => {
    const expenseWord = expensesCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,00.00');
    
    return (
        <div className="page-header">
            <div className="content-container">
                <h2 className="page-header__title">You are viewing <span>{expensesCount}</span> {expenseWord} totalling <span>{formattedExpensesTotal}</span></h2>
                <div className="page-header__actions">
                <Link to="create" className="button">Add Expense</Link>
                </div>
            </div>
        </div>
    )};

    const mapStateToProps = (state) => {
        const filteredExpenses  = selectExpenses(state.expenses, state.filters)
        return {
            expensesCount: filteredExpenses.length,
            expensesTotal: totalExpenses(filteredExpenses)
        }
    }

export default connect(mapStateToProps)(ExpenseSummary);

