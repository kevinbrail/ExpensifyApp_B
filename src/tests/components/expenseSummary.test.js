import React from 'react';
import { shallow } from 'enzyme';
import {ExpenseSummary} from '../../components/expenseSummary';

test('Should correctly render summary for multiple expenses', () => {
    const wrapper = shallow(<ExpenseSummary expensesCount={1} expensesTotal={3000} />);
    expect(wrapper).toMatchSnapshot();
})

test('Should correctly render summary for one expense', () => {
    const wrapper = shallow(<ExpenseSummary expensesCount={3} expensesTotal={12000} />);
    expect(wrapper).toMatchSnapshot();
})