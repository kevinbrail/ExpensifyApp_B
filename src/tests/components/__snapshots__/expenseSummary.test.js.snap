// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render summary for multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h2
      className="page-header__title"
    >
      You are viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        $30.00
      </span>
    </h2>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`Should correctly render summary for one expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h2
      className="page-header__title"
    >
      You are viewing 
      <span>
        3
      </span>
       
      expenses
       totalling 
      <span>
        $120.00
      </span>
    </h2>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
